cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc

```diff
@@ -54,50 +73,93 @@ CALL db.indexes
 
 //table
 
-== Index-backed property-lookup
+=== Index-backed property-lookup
 
 Now that we have a model of movies, actors and directors we can ask a question like 'find persons with the name Tom that acted in a movie':
 
 [source, cypher]
 ----
-MATCH (p:Person)-[:ACTED_IN]->(m:Movie) USING INDEX p:Person(name) WHERE p.name STARTS WITH 'Tom'
+MATCH (p:Person)-[:ACTED_IN]->(m:Movie) WHERE p.name STARTS WITH 'Tom'
 RETURN p.name, count(m)
 ----
 
 //table
 
 We have asked the database to return all the actors with the first name 'Tom'. There are three of them _'Tom Cruise'_, _'Tom Skerritt'_ and _'Tom Hanks'_.
-In previous versions of Neo4j, the final clause `RETURN p.name` would cause the database to take the node `p` and lookup its properties and return the value of the property `name`.
-In Neo4j 3.5 we can now support string properties in the native index, and this index is capable of returning the indexed property value together with the indexed nodes.
+In previous versions of Neo4j, the final clause `RETURN p.name` would cause the database to take the node `p` and lookup its properties and return the value
+of the property `name`.
+In Neo4j 3.5 we can now leverage the fact that indexes store the property values.
+In this case, it means that the names can be looked up directly from the index.
 This allows Cypher to avoid the second call to the database to find the property, which can save time on very large queries.
+If we profile the above query, we see that the `NodeIndexScan` in the Variables column contains `cached[p.name]`, which means that `p.name` is retrieved from the index.
+We can also see that the `Projection` has no DB Hits, which means it does not have to access the database again.
+
+[source, cypher]
+----
+PROFILE
+MATCH (p:Person)-[:ACTED_IN]->(m:Movie) WHERE p.name STARTS WITH 'Tom'
+RETURN p.name, count(m)
+----
+
+//profile
+
+If we change the query, such that it cannot use an index any more, we will see that there will be no `cached[p.name]` in the Variables, and that the
```

Comment:
```suggestion
If we change the query, such that it can no longer use an index, we will see that there will be no `cached[p.name]` in the Variables, and that the
```